#VRML_SIM R2022a utf8
WorldInfo {
  info [
    "The MyBot robot, the robot used in the tutorial of Webots"
  ]
  title "MyBot"
}
Viewpoint {
  orientation -0.35264888760051216 -0.1327718713458781 0.9262885037899558 3.811098975789829
  position 0.7209192160858507 0.8667798472349237 1.1612477939605483
}
TexturedBackground {
}
TexturedBackgroundLight {
}
RectangleArena {
}
Robot {
  translation 0 -0.3 0
  rotation 0 0 1 1.2272803122788083
  children [
    GPS {
    }
    Transform {
      translation 0 0 0.0415
      rotation 0 0 1 -1.570796
      children [
        Shape {
          appearance PBRAppearance {
            baseColor 0.0820075 0.364731 0.8
            roughness 1
            metalness 0
          }
          geometry DEF BODY Cylinder {
            height 0.08
            radius 0.045
          }
        }
      ]
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 3.6147199999999393
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "left wheel motor"
          maxVelocity 11
        }
        PositionSensor {
          name "left wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 0.04500000000000001 0.025
        rotation -0.7053186656728345 0.7053188468952841 0.0710697127319292 3.2834935777599275
        children [
          DEF WHEEL Transform {
            rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
            children [
              Shape {
                appearance PBRAppearance {
                  baseColor 1 0 0
                  roughness 1
                  metalness 0
                }
                geometry Cylinder {
                  height 0.01
                  radius 0.025
                }
              }
            ]
          }
        ]
        name "left wheel"
        boundingObject USE WHEEL
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 15.13791999999982
        axis 0 1 0
        anchor 0 0 0.025
      }
      device [
        RotationalMotor {
          name "right wheel motor"
          maxVelocity 11
        }
        PositionSensor {
          name "right wheel sensor"
        }
      ]
      endPoint Solid {
        translation 0 -0.04500000000000006 0.025
        rotation -0.6982279850709701 0.6982279239899609 -0.15797293130281023 2.828236351519406
        children [
          USE WHEEL
        ]
        name "right wheel"
        boundingObject USE WHEEL
      }
    }
    DistanceSensor {
      translation 0.042 0.02 0.063
      rotation 3.38385e-07 7.19235e-09 -1 -0.49920530717958655
      children [
        DEF SENSOR Transform {
          rotation 0.577656895078166 0.5771968951617173 -0.5771968951617173 -2.093935
          children [
            Shape {
              appearance PBRAppearance {
                baseColor 0.975691 0.981481 0.0252992
                roughness 1
                metalness 0
              }
              geometry Cylinder {
                height 0.004
                radius 0.008
              }
            }
          ]
        }
      ]
      name "ds_left"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    DistanceSensor {
      translation 0.042 -0.02 0.063
      rotation -3.14419e-07 -7.19235e-09 1 -0.5007953071795859
      children [
        USE SENSOR
      ]
      name "ds_right"
      lookupTable [
        0 1024 0
        0.05 1024 0
        0.15 0 0
      ]
      numberOfRays 2
      aperture 1
    }
    Shape {
      appearance PBRAppearance {
        baseColor 0.8 0.8 0.8
        baseColorMap ImageTexture {
          url [
            "https://raw.githubusercontent.com/cyberbotics/webots/R2022a/projects/samples/mybot/worlds/textures/mybot.png"
          ]
        }
        roughness 0.5
        metalness 0
      }
      geometry IndexedFaceSet {
        coord Coordinate {
          point [
            0.041 -0.015 0.038
            0.041 -0.015 0.023
            0.0455 0 0.023
            0.041 0.015 0.023
            0.041 0.015 0.038
            0.0455 0 0.038
          ]
        }
        texCoord TextureCoordinate {
          point [
            0 0
            0.5 0
            1 0
            1 1
            0.5 1
            0 1
          ]
        }
        coordIndex [
          0, 1, 2, 5, -1, 5, 2, 3, 4, -1
        ]
        texCoordIndex [
          5, 0, 1, 4, -1, 4, 1, 2, 3, -1
        ]
        creaseAngle 0.9
      }
    }
  ]
  name "MyBot"
  boundingObject Transform {
    translation 0 0 0.0415
    rotation 0 0 1 -1.570796
    children [
      USE BODY
    ]
  }
  controller "LawnBot"
}
